py/dml/serialize.py

```diff
@@ -291,7 +320,7 @@ def map_dmltype_to_attrtype(site, dmltype):
 # * One valid type string can never be a prefix of another valid type string
 # * Compound types can be parameterized by at most one type, and the parameter
 #   type's signature is a suffix of the compound type's signature.
```

Comment:
please explain the `is_for_serialization` arg

Reply:
Only to you or also through a comment?
`is_for_serialization` denotes whether the `type_signature` call is used to generate the name for the serialization function for the type, or for the deserialization function. The distinction for trait references is important since the serialization function for traitrefs can be shared, but the deserialization function cannot: so for serialization the type signature for traitrefs is only `T`, but for deserialization it's `T<template name>`.

Reply:
For the comment.